react/features/base/tracks/middleware.js

```diff
@@ -94,6 +96,8 @@ MiddlewareRegistry.register(store => next => action => {
             createFakeScreenShareParticipant(store, action);
         }
 
+        !local && jitsiTrack.on(JitsiTrackEvents.TRACK_STREAMING_STATUS_CHANGED, (jitsiTrack, streamingStatus) => store.dispatch(trackStreamingStatusChanged(jitsiTrack, streamingStatus)));
```

Comment:
So we actually don't want to do it this way because it means registering for events to every remote track. William has added optimizations in LJM around turning the track streaming status tracking off if there are no event listeners registered to it. Ideally, the UI component that's making use of the track streaming status should register the listener and use the status locally.

What's the specific case that is to be fixed by this PR?

Reply:
This is to fix video not shown due to connectStatusIcon not mounted when in mobile view or when config.connectionIndicators.disabled is true.

Reply:
Large video and thumbnails all make use of the track streaming status. I tried to do it in both UI components and feel weird. I guess it's a tradeoff between cleaner UI code vs. better performance. How much performance improvement are we talking about here?
Do we have an app state of whether a participant's thumbnail or large video is visible?

Reply:
does it feel weird in UI components because track streaming status is written to redux? would it feel less weird if the state was kept locally in the component?

Reply:
Track streaming status is problematic in large conferences we're talking about 500 participants. It was specifically decided to do the optimization where the status is only calculated if the track is displayed (when there's UI component displayed that needs the status).

Reply:
> Track streaming status is problematic in large conferences we're talking about 500 participants. It was specifically decided to do the optimization where the status is only calculated if the track is displayed (when there's UI component displayed that needs the status).

Understood

Reply:
> does it feel weird in UI components because track streaming status is written to redux? would it feel less weird if the state was kept locally in the component?

1. It makes sense to me to have track streaming status in redux because a) it is an obvious track property and track should be and is in redux, b) there can be multiple components that want to use it. 
2. Because track streaming status is such an obvious track property, it should be managed in the life cycle of a track, I.e. track actions, instead of UI components that use it. It’s like, state should control view, not the other way around. 
3. To optimize, we can use both track actions and visibility actions (I don’t know if there is such a thing and how the visibility is controlled), instead of asking every UI component to touch the state directly. Ideally the visibility state controls the render of UI components. 
4. The code will be harder to read and reason about and maintain if we do it in the UI components.

Reply:
After some tradeoff, I decide to handle it in LargeVideoManager.js and Thumbnail.js. Because it's easier to do a quick fix this way than to modify the visibility state/action to fit for this change. 
Yet still, the proper way in the long term is still to use state/action to control track streaming status, not UI components.

Reply:
I disagree on that the track streaming status should be in Redux. We're trying to optimize the usage of the streaming status and have it tracked only if there are UI components that require it. That use case is best solved with functional components and react hooks which create local state for the UI component and make it easy to dispose the listener when the UI component is removed from the DOM. It will be much more complicated and error prone trying to track components visibility states and add/remove listener.

https://reactjs.org/docs/hooks-intro.html